### PR TITLE
Adding new options to auto battle

### PR DESCRIPTION
**Pull request by dlsantana**
Changes to auto battle:

- Added list of pokémon to avoid to battle.yml.
- Added list of pokémon to target to battle.yml.
- Added HP threshold setting to battle.yml.
- Added behavior options for when the lead pokémon is unable to battle. Now the user can make the bot go to manual mode.
- Fixed issue that made pokémon keep battling even when its HP was below the minimum threshold.

### CODE REVIEW

**Comment by pekempy** [comment]
Hey, please lint your code using [black](https://github.com/psf/black) so that the GitHub actions passes.

`pip install black`
`black . --line-length 120`

**Comment by dlsantana** [comment]
> Hey, please lint your code using [black](https://github.com/psf/black) so that the GitHub actions passes.
> 
> `pip install black` `black . --line-length 120`

Thank you for the tip!
Did that and the test passed now.

File: modules/battle.py [hunk 1]
```diff
@@ -574,23 +584,31 @@ def determine_battle_menu_action(self):
...
-                self.choice = "flee"
-                self.idx = -1
-            else:
-                self.choice = "switch"
-                self.idx = mon_to_switch
+        elif not check_mon_can_battle(self.current_battler):
+            match context.config.battle.lead_cannot_battle_action:
+                case "flee":
+                    self.choice = "flee"
+                    self.idx = -1
```
[the remark this answers is not in the document]
Comment (dlsantana) [reply modules/battle.py]: Added check for trainer battles as recommended

File: modules/battle.py [hunk 2]
```diff
@@ -1212,26 +1231,45 @@ def check_for_level_up(old_party: list[Pokemon], new_party: list[Pokemon], level
     return leveled_mon
 
 
-def can_battle_happen(check_lead_only: bool = False) -> bool:
+def can_battle_happen(battle_opponent: BattleOpponent, check_lead_only: bool = False) -> bool:
```
[the remark this answers is not in the document]
Comment (dlsantana) [reply modules/battle.py]: Added separate function to check whether an opponent pokémon should be battled as recommended

**Review by 40Cakes — APPROVED** [review]

**Comment by 40Cakes** [comment]
Ah I approved and thought this was all good to merge sorry, but seems stuck til @hanzi resolves the requested changes..

```
1 change requested
@hanzi
hanzi requested changes 
```

**Pull request merged**